[19.0] [MIG] queue_job_batch: migrate + tests

tishmen opened the pull request:
Scope
- queue_job_batch, test_queue_job_batch

Depends on
- [x] https://github.com/OCA/queue/pull/840

Summary
- Migrate to Odoo 19; tests included. No functional changes beyond compatibility.

Temporary references
- test-requirements.txt adds temporary dependencies to the base PR for CI/runboat.
  This will be removed once the base PR merges.

Pre-commit
- Ran locally; auto-fixes committed.

Tests
- CI will run the suite using the referenced base PR. Local isolated runs may skip
  test_queue_job_batch when queue_job/test_queue_job are not present.

hoangtrann reviewed (commented):
Everything looks good to me at first glance, I'll run the test locally and circle back

hoangtrann commented on queue_job_batch/security/security.xml:
```diff
@@ -3,7 +3,8 @@
         <record id="group_queue_job_batch_user" model="res.groups">
             <field name="name">Job Queue Batch User</field>
-            <field name="category_id" ref="queue_job.module_category_queue_job" />
+            <!-- Odoo 19: classify via privilege -->
```
should be removed

hoangtrann commented on queue_job_batch/views/queue_job_batch_views.xml:
```diff
@@ -64,11 +64,8 @@
-                    groups="base.group_multi_company"
-                    widget="selection"
-                />
+                <!-- Odoo 19: no widget override in search; keep Many2one as-is -->
```
to be removed

hoangtrann commented on queue_job_batch/views/queue_job_batch_views.xml:
```diff
@@ -96,18 +93,18 @@
-                    />
-                </group>
+                <separator />
+                <!-- Odoo 19: no <group expand> in search; declare group-by filters inline -->
```
to be removed

hoangtrann reviewed (changes requested):
Please update the test-requirements.txt and check if tests are now passing too

tishmen committed:
[19.0][MIG] queue_job_batch: migrate

Scope: queue_job_batch, test_queue_job_batch

tishmen commented:
Hi @hoangtrann, thanks for the review.

  - Removed the migration-related inline comments you pointed out.
  - Dropped the temporary test-requirements.txt (git URL refs) and restored requirements.txt.
  - Rebased on latest 19.0, then squashed everything into a single [19.0][MIG] queue_job_batch: migrate commit.

  CI is green now. @sbidoul

len-foss reviewed (changes requested):
I'm quite confused, but AFAICT you need https://github.com/lambdao-dev/queue/commit/21c406209ad2c378d76daa6d86039f06ed3063d6 to have the migration working.

qgroulard commented:
Hi, could you please also include this https://github.com/OCA/queue/pull/908 ?
